presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoResultSet.java

```diff
@@ -1761,35 +1758,17 @@ private static Number toNumber(Object value)
         private final StatementClient client;
         private final Consumer<QueryStats> progressCallback;
         private final WarningsManager warningsManager;
-        private final boolean isQuery;
 
         private ResultsPageIterator(StatementClient client, Consumer<QueryStats> progressCallback, WarningsManager warningsManager)
         {
             this.client = requireNonNull(client, "client is null");
             this.progressCallback = requireNonNull(progressCallback, "progressCallback is null");
             this.warningsManager = requireNonNull(warningsManager, "warningsManager is null");
-            this.isQuery = isQuery(client);
-        }
-
-        private static boolean isQuery(StatementClient client)
-        {
-            String updateType;
-            if (client.isRunning()) {
-                updateType = client.currentStatusInfo().getUpdateType();
-            }
-            else {
-                updateType = client.finalStatusInfo().getUpdateType();
-            }
-            return updateType == null;
         }
 
         @Override
         protected Iterable<List<Object>> computeNext()
         {
-            if (isQuery) {
-                // Clear the warnings if this is a query, per ResultSet javadoc
-                warningsManager.clearWarnings();
```

Comment:
Currently the test pulls everything into single bag of warnings (with `getStatementWarnings` method) and so we're not testing when warnings are visible and when they get cleared.

Would be good to have explicit test for the tested behavior. Like:

```
try (statement = ..) {
   try (rs = ...) {
      rs.next();
      check statement.getWarnings
      check rs.getWarnings
      rs.next();
      check statement.getWarnings
      check rs.getWarnings
  }
}
```

Reply:
Agreed, see comment above